packages/did-ethr-resolver/src/implementations/resolver.ts

```diff
@@ -54,10 +54,12 @@ class Resolver implements IResolver {
    * Constructor
    *
    * Settings have to be passed to construct resolver
-   * @param {IResolverSettings} settings
+   * @param {RegistrySettings} settings
    */
-  constructor(provider: providers.Provider, settings: RegistrySettings) {
-    this._provider = provider;
+  
+  // constructor(provider: providers.Provider, settings: RegistrySettings) {
+  constructor(privateKey: string, settings: RegistrySettings, providerUrl?: string) {
```

Comment:
Shouldn't need a `privateKey` just to resolve DID documents 🤔 
```suggestion
  constructor(settings: RegistrySettings, providerUrl?: string) {
```